### PR TITLE
Add ability for users/admin to delete reviews

### PR DESCRIPTION
**Pull request by kstrikis**

### CODE REVIEW

**Comment by kstrikis** [comment]
Demo: https://www.loom.com/share/bb6c6458e31345a0a00dcbd9447a6684

File: client/src/components/DogParkShow.js [hunk 1]
```diff
@@ -34,9 +41,48 @@ const DogParkShow = (props) => {
         return <p className="tag-cloud-individual-tag">{tag}</p>
     })
 
+    const deleteReview = async (reviewId) => {
+        try {
+            const response = await fetch(`/api/v1/parks/${parkId}/reviews/${reviewId}`, { method: "DELETE"})
```
Comment (dovakeen118) [inline client/src/components/DogParkShow.js]: when deleting a review we only need to know the id of the review to delete - parks is not needed in the delete request
```js
`/api/v1/reviews/${reviewId}`
```

File: client/src/components/DogParkShow.js [hunk 2]
```diff
@@ -34,9 +41,48 @@ const DogParkShow = (props) => {
...
+        let deleteButton = ""
+        if (isAdmin || review.userId === userId) {
+            deleteButton =
+                <button
+                    className="button"
+                    onClick={(event) => {
+                    handleOnClickDeleteReview(event, review.id)}}>
+                        Delete review
+                </button>
+        }
```
Comment (dovakeen118) [inline client/src/components/DogParkShow.js]: consider moving this logic to the ReviewTile to clean up this DogParkShow component

File: client/src/components/DogParkShow.js [hunk 3]
```diff
@@ -34,9 +41,48 @@ const DogParkShow = (props) => {
...
         return (
-            <ReviewTile key={review.id} {...review}/>
+            <>
+                <ReviewTile key={review.id} {...review}/>
+                {deleteButton}
+            </>
```
Comment (dovakeen118) [inline client/src/components/DogParkShow.js]: the review tile is the top level element - remove fragment and define all HTML in the same location (ReviewTile)

File: server/src/routes/api/v1/parksReviewRouter.js [hunk 4]
```diff
@@ -26,4 +26,14 @@ parksReviewRouter.post("/", async (req, res) => {
     }
 })
 
+parksReviewRouter.delete("/:reviewId", async (req, res) => {
+    const { reviewId } = req.params
+    try {
+        await Review.query().deleteById(reviewId)
+        res.status(200).json( {message: "Review deleted"} )
+    } catch(err) {
+        res.status(500).json( {errors: err.message} )
```
Comment (dovakeen118) [inline server/src/routes/api/v1/parksReviewRouter.js]: lines 33 and 35, add space padding inside braces, not parentheses

```js
({ message: "Review deleted" })
```

**Review by dovakeen118 — COMMENTED** [review]
small comments, then merge!

**Pull request merged**